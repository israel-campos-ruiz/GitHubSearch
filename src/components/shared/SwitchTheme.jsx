import React, { useRef } from "react";
import "../../styles/shared/SwitchTheme.css"

const SwitchTheme = ({ checked, setDarkMode, setChecked }) => {
  const ref = useRef(null);
  const handleChange = () => {
    setChecked(ref?.current?.checked);
    setDarkMode(ref?.current?.checked);
  };

  return (
    <div className="dark-mode">
      <input
        ref={ref}
        onChange={handleChange}
        type="checkbox"
        checked={checked}
        className="checkbox"
        id="checkbox"
      />
      <label className="switch" htmlFor="checkbox"></label>
    </div>
  );
};

export default SwitchTheme;
