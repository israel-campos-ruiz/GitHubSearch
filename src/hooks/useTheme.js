import { useState, useEffect } from 'react';

const useTheme = () => {
    const [darkMode, setDarkMode] = useState(false);
    const [checked, setChecked] = useState(false);

    const changeMediaQuery = (mq) => {
        setDarkMode(mq.matches);
        setChecked(mq.matches);
    }

    useEffect(() => {
        const mq = window.matchMedia('(prefers-color-scheme:dark)')
        mq.addListener(changeMediaQuery);
        setDarkMode(mq.matches)
        setChecked(mq.matches)
    },[])

    return{
        darkMode,
        checked,
        setDarkMode,
        setChecked
    }
}

export default useTheme
