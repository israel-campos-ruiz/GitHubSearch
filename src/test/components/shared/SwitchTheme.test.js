import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import SwitchTheme from '../../../components/shared/SwitchTheme';
import useTheme from '../../../hooks/useTheme';
import { renderHook, act } from '@testing-library/react-hooks'
describe('Test in the component  <SwitchTheme/>',() =>{

    test('shold be render a component', () => {
       const wrapper = shallow(<SwitchTheme/>);
       expect(wrapper).toMatchSnapshot()
   });

   test('the darkmode should be false in the light-mode', () => {
   const {result} =  renderHook(() =>  useTheme())
   const {darkMode} = result.current;
   expect(darkMode).toBe(false)
   });
   test('the function setChecked to be false', () => {
   const {result} =  renderHook(() =>  useTheme())
   const {setChecked, checked} = result.current;
    act(()=>{
        setChecked()
    })
    expect(checked).toBe(false)
    
   })
});