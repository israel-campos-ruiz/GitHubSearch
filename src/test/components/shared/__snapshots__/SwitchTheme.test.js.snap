// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in the component  <SwitchTheme/> shold be render a component 1`] = `
<div
  className="dark-mode"
>
  <input
    className="checkbox"
    id="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="switch"
    htmlFor="checkbox"
  />
</div>
`;
